pkg/chunk/aws/dynamodb_storage_client.go

```diff
@@ -865,5 +867,24 @@ func awsSessionFromURL(awsURL *url.URL) (client.ConfigProvider, error) {
 		return nil, err
 	}
 	config = config.WithMaxRetries(0) // We do our own retries, so we can monitor them
+	config = config.WithHTTPClient(&http.Client{Transport: defaultTransport})
 	return session.NewSession(config)
 }
+
+// Copy-pasted http.DefaultTransport
+var defaultTransport http.RoundTripper = &http.Transport{
+	Proxy: http.ProxyFromEnvironment,
+	DialContext: (&net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+		DualStack: true,
```

Comment:
This is deprecated. Given now it's enabled by default, this option can be removed.

Reply:
Interesting - why doesn't the linter show it as deprecated?

Reply:
I have no idea. I spent an hour on it, without any result. I can't get the deprecated check working (which should be done by `staticcheck`).

Reply:
I removed the line.